modules/regex/SCsub

```diff
@@ -1,7 +1,53 @@
 #!/usr/bin/env python
 
 Import('env')
+Import('env_modules')
 
-env.add_source_files(env.modules_sources, "*.cpp")
+env_regex = env_modules.Clone()
 
-Export('env')
+thirdparty_dir = "#thirdparty/pcre2/src/"
+thirdparty_sources = [
+    "pcre2_auto_possess.c",
+    "pcre2_chartables.c",
+    "pcre2_compile.c",
+    "pcre2_config.c",
+    "pcre2_context.c",
+    "pcre2_dfa_match.c",
+    "pcre2_error.c",
+    "pcre2_find_bracket.c",
+    "pcre2_jit_compile.c",
+    "pcre2_maketables.c",
+    "pcre2_match.c",
+    "pcre2_match_data.c",
+    "pcre2_newline.c",
+    "pcre2_ord2utf.c",
+    "pcre2_pattern_info.c",
+    "pcre2_serialize.c",
+    "pcre2_string_utils.c",
+    "pcre2_study.c",
+    "pcre2_substitute.c",
+    "pcre2_substring.c",
+    "pcre2_tables.c",
+    "pcre2_ucd.c",
+    "pcre2_valid_utf.c",
+    "pcre2_xclass.c",
+]
+thirdparty_sources = [thirdparty_dir + file for file in thirdparty_sources]
+thirdparty_flags = ["-DPCRE2_STATIC", "-DHAVE_CONFIG_H", "-DSUPPORT_JIT"]
+
+env_regex.Append(CPPPATH=[thirdparty_dir])
+env_regex.Append(CPPFLAGS=thirdparty_flags)
+env_regex.Append(CPPFLAGS=["-DPCRE2_CODE_UNIT_WIDTH=0"])
+env_regex.add_source_files(env.modules_sources, "*.cpp")
+
+def pcre2_builtin(width):
+    env_pcre2 = env_modules.Clone()
+    env_pcre2["OBJSUFFIX"] = "_" + width + env_pcre2["OBJSUFFIX"]
+    env_pcre2.Append(CPPPATH=[thirdparty_dir])
+    env_pcre2.add_source_files(env.modules_sources, thirdparty_sources)
+    env_pcre2.Append(CPPFLAGS=thirdparty_flags)
+    env_pcre2.Append(CPPFLAGS=["-DPCRE2_CODE_UNIT_WIDTH=" + width])
+
+if (env['builtin_pcre2'] != 'no'):
```

Comment:
So I've moved as much as possible into the `if` block to allow external linking.

On X11, `builtin_pcre2` is still by default set to yes, as it's not available on Ubuntu Trusty (and thus Travis-CI). Is there a way to automatically select yes/no if the native lib is not-found/found?